[8.0] FileCatalog: add function to get user metadata for list of LFNs

andresailer opened the pull request:
I need a function to get the user metadata for a list of LFNs. There are functions to get this information based on metadata, but I don't / can't do that, and for efficiency I need a bulk method. 

I am not sure I am missing some pieces, but this works for my purposes.

I added the getFileDetailsPublic to add a checkPathPermission before getting details, similar to getFileMetadata, but I am not sure if this is necessary or not.

Since I can hotfix this on my servers it is not urgent at all

BEGINRELEASENOTES

*DMS
NEW: FileCatalogHandler: add function export_getFileDetails to get the (user) metadata for a list of LFNs

ENDRELEASENOTES

fstagni commented on src/DIRAC/DataManagementSystem/Service/FileCatalogHandler.py:
```diff
@@ -261,6 +261,12 @@ def export_getFileMetadata(self, lfns):
 
+    types_getFileDetails = [[list, dict, str]]
+
+    def export_getFileDetails(self, lfns):
```
Out of consistency, would you call this 
```suggestion
    def export_getFileDetailsPublic(self, lfns):
```

But this sounds better:

```suggestion
    def export_getPublicFileDetails(self, lfns):
```

andresailer replied:
Well, depends of the consistency with what.

I went for consistency with `export_getFileMetadata`, and the `Public` is supposed to point out that that one has the permission check, while the non public one doesn't have the permission check.
And the file details aren't public, I think? I don't know who can see which meta data to be honest.

Maybe getFileUserMetadata would be better anyway, because `Details` is very vague?

fstagni replied:
I meant consistency with the method name used at DB level.

andresailer replied:
But the exported function should be clearer for the user, and not worry about the implementation details underneath. So, `export_getFileUserMetadata` might be better?

fstagni replied:
Fine for me!

andresailer replied:
Ahh, that function name already exists, but is not working for bulk :/

fstagni replied:
@andresailer if you sort this one out I can merge it this afternoon (and make a release with it just after).

andresailer replied:
Just use a different name that doesn't exist yet, or something more involved?

fstagni replied:
Just use a different name

andresailer replied:
Went back to `export_getFileDetails`

andresailer committed:
feat(FC): Add function getFileDetails to get user metadata for list of lfns

add DB internal getFileDetailsPublic to add permission check before getFileDetails

andresailer committed:
fix(FC): correct docstring for getDirectoryMetadata
